flutter-studio/src/io/flutter/profiler/FlutterStudioMonitorStageView.java

```diff
@@ -409,30 +417,23 @@ public void received(Isolate response) {
           LibraryRef ref = iterator.next();
 
           if (ref.getName().length() > 0) {
-            // Non-empty string is the library name (use it for key)
-            if (ref.getName().startsWith("dart.")) {
-              dartLibraries.put(ref.getName(), ref);
-            }
-            else if (ref.getUri().startsWith("file:///")) {
-              // Is user code (not in a package or library)
-              // TODO(terry): Need to store local file names for each libraryRef we'll always display classes from user code.
-              allLibraries.put(ref.getName(), ref);
-            }
-            else {
-              if (ref.getUri().startsWith(DART_LIBRARY_PREFIX)) {
-                // Library named 'nativewrappers' but URI is 'dart:nativewrappers' is a Dart library.
-                dartLibraries.put(ref.getName(), ref);
-              }
-              else {
-                allLibraries.put(ref.getUri(), ref);
-              }
-            }
+            rememberLibrary(ref.getName(), ref);
+          } else {
+            // No unique name to display use the Uri and don't display in list of known library/packages.
+            rememberLibrary(PREFIX_LIBRARY_NAME_HIDDEN + ref.getUri(), ref);
           }
         }
-        allLibraries.put("dart:*", null);   // All Dart libraries are in this entry.
+        allLibraries.put(ALL_DART_LIBRARIES, null);       // All Dart libraries are in this entry.
+        allLibraries.put(ALL_FLUTTER_LIBRARIES, null);    // All Flutter libraries are in this entry.
 
+        Set<String> displayedLibraries = new TreeSet<String>();
```

Comment:
nit: can write new TreeSet<>() on the RHS

Reply:
Yep, done.